instrumentation/log4j/log4j-appender-2.17/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/log4j/appender/v2_17/Log4jHelper.java

```diff
@@ -86,11 +99,53 @@ public static void capture(
         contextData,
         threadName,
         threadId,
+        () -> location != null ? location : getLocation(loggerClassName),
         Context.current());
     builder.setTimestamp(Instant.now());
     builder.emit();
   }
 
+  private static StackTraceElement getLocation(String loggerClassName) {
+    if (stackTraceMethodHandle == null) {
+      return null;
+    }
+
+    try {
+      return (StackTraceElement) stackTraceMethodHandle.invoke(loggerClassName);
+    } catch (Throwable exception) {
+      return null;
+    }
+  }
+
+  private static MethodHandle getStackTraceMethodHandle() {
+    // since 2.9.0
+    Class<?> stackTraceClass = null;
+    try {
+      stackTraceClass = Class.forName("org.apache.logging.log4j.util.StackLocatorUtil");
+    } catch (ClassNotFoundException exception) {
+      // ignore
+    }
+    if (stackTraceClass == null) {
+      try {
+        stackTraceClass = Class.forName("org.apache.logging.log4j.core.impl.Log4jLogEvent");
```

Comment:
The stack trace class is not the same following the Log4j version? If so, could you please add comments to explain this?  

It's perhaps the intent of the comment added at line 121.? If so, could you please move this comment around the line 124?